tool/README.md

```diff
@@ -1,130 +1,159 @@
-- flutter run an application
-- connect to the running app from devtools, and verify that the pages
-  generally work, and there are no exceptions in the chrome devtools log
+- `flutter run` an application
+- connect to the running app from DevTools, verify that the pages
+  generally work and that there are no exceptions in the chrome devtools log
 
 #### Publish the packages
 
 ```shell
 ./tool/pub_publish.sh
 ```
 
-#### Revert the change to .gitignore and pubspec files
+#### Revert changes caused by the `publish.sh` script
 ```shell
-git checkout .gitignore
-git checkout packages/*/pubspec.yaml
-```
-
-Update the version of DevTools referenced by
-https://github.com/flutter/flutter/blob/8c4538618f81a42e45cdb03c6a204f6e7bcf81b3/dev/devicelab/lib/tasks/perf_tests.dart#L1448
-
-with a CL like
-https://github.com/flutter/flutter/pull/81869
-
-to avoid the issue described by https://github.com/flutter/flutter/issues/81552
-Once DevTools is deployed as part of the Dart SDK we should be able to eliminate this step.
+git checkout .
 
+git clean -f -d
+```
 
 #### Create the tag for this release and push to the remote repository.
 This script will automatically determine the version from the `packages/devtools/pubspec.yaml` so there
 is no need to manually enter the version.
 ```shell
 tool/tag_version.sh
 ```
+
+#### Upload the DevTools binary to CIPD
+
+Checkout the version of DevTools you just tagged (for example, `2.7.0`):
+```shell
+git checkout v2.7.0
+
+Note: switching to 'v2.7.0'.
+
+You are in 'detached HEAD' state. You can look around, make experimental
+changes and commit them, and you can discard any commits you make in this
+state without impacting any branches by switching back to a branch.
+
+If you want to create a new branch to retain commits you create, you may
+do so (now or later) by using -c with the switch command. Example:
+
+  git switch -c <new-branch-name>
+
+Or undo this operation with:
+
+  git switch -
+
+Turn off this advice by setting config variable advice.detachedHead to false
+
+HEAD is now at 8881a7ca Update release scripts (#3426)
+```
+
+Copy the commit hash from the bottom of the CLI output (in the example above, `8881a7ca`).
+
+Using the [update.sh](https://github.com/dart-lang/sdk/blob/master/third_party/devtools/update.sh)
+script at https://github.com/dart-lang/sdk/tree/master/third_party/devtools, build DevTools at the 
+given git commit hash, and upload the binary to CIPD.
+
+```shell
+sdk/third_party/devtools/update.sh 8881a7ca
+```
+
+#### Update the DevTools hash in the Dart SDK
+
+Update the `devtools_rev` entry in the Dart SDK 
+[DEPS file](https://github.com/dart-lang/sdk/blob/master/DEPS)
+with the git commit hash you just built DevTools from (this is
+the id for the CIPD upload in the previous step). See this 
+[example CL](https://dart-review.googlesource.com/c/sdk/+/215520).
+
+Now verify that running `dart devtools` launches the version of DevTools you just released.
```

Comment:
You'll need to build the SDK to check this:

```
$ cd ~/dart-sdk/sdk
$ gclient sync -D
$ ./tools/build.py -mrelease -ax64 create_sdk // create_platform_sdk is faster but doesn't build web tooling
$ out/ReleaseX64/dart-sdk/bin/dart devtools // On OSX replace 'out' with 'xcodebuild'
```

Reply:
added.